src/components/AgentAuthority/LicenseCredDef/LicenseCredDef.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Get or create the credential definitions for the
+ * license issued by the authority
+ */
+import { useEffect } from 'react'
+
+import { AUTHORITY_LABEL, LICENSE_SCHEMA_NAME } from '../../../utils/env.js'
+import useGetCredDefinitionsCreated from '../../../interface/hooks/use-get-cred-definitions-created'
+import usePostSchemas from '../../../interface/hooks/use-post-schemas'
+import usePostCredentialDefinitions from '../../../interface/hooks/use-post-credential-definitions'
+import Error from '../../Common/Misc/Error'
+
+export default function LicenseCredDef({ origin, setLicenseCredDefId }) {
+  const [errorDefinitionsCreated, startFetchHandlerDefinitionsCreated] =
+    useGetCredDefinitionsCreated()
+
+  const [errorPostSchema, startPostSchema] = usePostSchemas()
+  const [errorPostCredDefs, startPostCredDef] = usePostCredentialDefinitions()
+
+  useEffect(() => {
+    startFetchHandlerDefinitionsCreated(origin, (credDefIds) => {
+      const licenseCredDefId = credDefIds.find((credDefId) =>
+        credDefId.includes(LICENSE_SCHEMA_NAME)
+      )
+      if (licenseCredDefId) {
+        setLicenseCredDefId(licenseCredDefId)
+      } else {
+        startPostSchema(origin, LICENSE_SCHEMA_NAME, (schemaId) => {
+          startPostCredDef(
+            origin,
+            schemaId,
+            AUTHORITY_LABEL,
+            setLicenseCredDefId
+          )
+        })
+      }
+    })
+  }, [
+    origin,
+    startFetchHandlerDefinitionsCreated,
+    startPostSchema,
+    startPostCredDef,
+    setLicenseCredDefId,
+  ])
+
+  return (
+    <Error
+      errors={[errorDefinitionsCreated, errorPostSchema, errorPostCredDefs]}
+    />
```

Comment:
I still don't like the duplicate `Error` component here and on [src/components/AgentAuthority/IssueLicense/IssueLicense.js](https://github.com/digicatapult/veritable-authority/pull/4/files#diff-f315495d2dae22e054082d8820f1d2dd9e729e345b01234e437154775c59efdf)

Reply:
Updated